pkg/provider/azure_vmssflex_cache.go

```diff
@@ -322,7 +322,7 @@ func (fs *FlexScaleSet) DeleteCacheForNode(nodeName string) error {
 		return err
 	}
 
-	cached, err := fs.vmssFlexVMCache.Get(vmssFlexID, azcache.CacheReadTypeDefault)
+	cached, err := fs.vmssFlexVMCache.GetWithDeepCopy(vmssFlexID, azcache.CacheReadTypeDefault)
```

Comment:
it will delete element in the map, so it should be non deep copy?

Reply:
Updated.